src/utilities/getProcessedImage.ts

```diff
@@ -0,0 +1,57 @@
+import { config } from './config';
+
+/**
+ *
+ * @param imageSrc - The Storyblok URL of the image
+ * @param crop - The dimension of the image crop (eg., "600 x 400")
+ * @param focus - The focal point of the image provided by the Storyblok Asset field (eg, "348x414:349x415")
+ * @param filters - Additional filters to apply to the image (eg., "blur(10)").
+ * To add multiple filters, separate them with a colon, eg., "blur(10):grayscale()"
+ * @returns The processed Storyblok image URL that is masked by our asset CDN
+ * @see {@link https://www.storyblok.com/docs/image-service} for more info on the Storyblok image service
+ */
+
+export const getProcessedImage = (
+  imageSrc: string,
+  crop: string = '',
+  focus: string = '',
+  filters: string = '',
+): string => {
+  const { imageService } = config;
+
+  // Get the width and the height from the crop dimension
+  const width = crop.split('x')[0];
+  const height = crop.split('x')[1];
+
+  // Adding '/m' at the end of the URL to automatically convert the image to webp if the browser supports it.
+  // Supports jpg, png (including transparent ones). Animated GIFs will retain the GIF format automatically.
+  let myParams = '/m';
+
+  // Start off the filters with a default image quality of 70% - works for webp
+  let myFilters = '/filters:quality(70)';
+
+  if (crop && width !== '0' && height !== '0') {
+    myParams += `/${crop}`;
+    /**
+     * Handle focus only if a crop dimension is provided and it has both width and height
+     * If a focal point is provided, add the focal point to the filters.
+     * If no focal point is provided, activate the "smart" face detection feature.
+     */
+    if (focus) {
+      myFilters += `:focal(${focus})`;
+    } else {
+      myParams += '/smart';
+    }
+  }
+
+  // Add any additional filters
+  if (filters) {
+    myFilters += `:${filters}`;
+  }
+
+  // The URL of the processed Storyblok image
+  const processedSbUrl = `${imageSrc}${myParams}${myFilters}`;
+  const maskedUrl = processedSbUrl.replace(imageService, `${config.assetCdn}a-us/`);
+
+  return maskedUrl;
+};
```

Comment:
This replaces `transformImage` in the SAA repo with updated Storyblok image service and support for the US Storyblok server, also a bunch of logic clean ups